### ISSUE TITLE
Debugging metrics.

### ISSUE THREAD

**Issue by arbaazkhan2** [waymo-research/waymo-open-dataset#320]
I am looking at 2 of my validation submissions and I am struggling to debug why the mAP scores on one of them are higher than the other despite having really high ADE and really high FDE. Images attached below; Any explanation would be useful. I have read through the description of all the metrics but I am still unsure why this is the case and would appreciate it if anyone could shed any light; 
![HighADE:HighmAP](https://user-images.githubusercontent.com/7920204/119872055-e8eca100-bed7-11eb-850b-d94f1d824174.png)
![LowADE:LowmAP](https://user-images.githubusercontent.com/7920204/119872060-e9853780-bed7-11eb-9852-651f9456d15c.png)

**Comment by arbaazkhan2** [comment]
A similar concern with miss rate. How can the miss rate be lower when the Min ADE and Min FDE are so much higher?

**Comment by scott-ettinger** [comment]
I agree those numbers are unexpected.  I don't have an immediate answer for what is going on there.  I will try to get some time to look at your submission file to see what is causing this.  The only thing I can think of is if your model does well on stationary and nearly stationary objects only, this could result in something like this.

**Comment by scott-ettinger** [comment]
Hi Arbaaz.  I took a quick look at your submissions.  In the first submission your model predicted stationary objects very well for cyclists at 3 seconds, but for all other trajectories it was not very accurate resulting in a high FDE score.  Because the mAP metric averages across trajectory type bins, stationary trajectories count for 1/8th of the mAP score.  This why your first model had a high mAP score for cyclists at 3 seconds.  Your second model did better in general resulting in a lower FDE score, but it did worse on the stationary trajectories, and did was not accurate enough on other trajectories to avoid misses on almost all trajectories.  This is why even though the FDE score is lower, the mAP score is worse.  I agree this is a bit counter intuitive.  Plotting the P/R curves for the trajectory bins helps a lot to understand this. We will consider adding these plots to the results in the future.
